crates/red_knot_python_semantic/src/types/property_tests.rs

```diff
@@ -0,0 +1,249 @@
+//! This module contains quickcheck-based property tests for `Type`s.
+//!
+//! These tests are feature-gated and disabled by default. You can run them using:
+//!
+//! ```sh
+//! cargo test -p red_knot_python_semantic --features property_tests types::property_tests
+//! ```
+//!
+//! You can control the number of tests (default: 100) by setting the `QUICKCHECK_TESTS`
+//! variable. For example:
+//!
+//! ```sh
+//! QUICKCHECK_TESTS=10000 cargo test …
+//! ```
+//!
+//! If you want to run these tests for a longer period of time, it's advisable to run them
+//! in release mode. As some tests are slower than others, it's advisable to run them in a
+//! loop until they fail:
+//!
+//! ```sh
+//! export QUICKCHECK_TESTS=100000
+//! while cargo test --release -p red_knot_python_semantic \
+//!   --features property_tests types::property_tests; do :; done
+//! ```
+
+use std::sync::{Arc, Mutex, MutexGuard, OnceLock};
+
+use super::tests::{setup_db, Ty};
+use super::Type;
+use crate::db::tests::TestDb;
+use crate::types::KnownClass;
+use crate::Db;
+use quickcheck::{Arbitrary, Gen};
+use quickcheck_macros::quickcheck;
+
+fn arbitrary_core_type(g: &mut Gen) -> Ty {
+    // We could select a random integer here, but this would make it much less
+    // likely to explore interesting edge cases:
+    let int_lit = Ty::IntLiteral(*g.choose(&[-2, -1, 0, 1, 2]).unwrap());
```

Comment:
Including `i64::MIN` and `i64::MAX` here also seems likely to uncover interesting edge cases?

Reply:
Ok. I wasn't sure about these. They seem more like red knot internal limitations. But if we add property tests for things like the type level `Literal[…]` math, adding those makes sense.

Reply:
Yeah, that's fair, it may not make sense to add these unless we add property tests for literal arithmetic.